src/test/mir-opt/issue_73223.main.PreCodegen.32bit.diff

```diff
@@ -47,60 +45,54 @@
           discriminant(_2) = 1;            // scope 0 at $DIR/issue-73223.rs:2:23: 2:30
           StorageLive(_3);                 // scope 0 at $DIR/issue-73223.rs:3:14: 3:15
           _3 = ((_2 as Some).0: i32);      // scope 0 at $DIR/issue-73223.rs:3:14: 3:15
-          _1 = _3;                         // scope 2 at $DIR/issue-73223.rs:3:20: 3:21
+          _1 = ((_2 as Some).0: i32);      // scope 2 at $DIR/issue-73223.rs:3:20: 3:21
```

Comment:
I'm surprised our DCE didn't get rid of `_3`, situations like this could explain why we don't see much gain in the perf runner, all the dead code is still passed to LLVM and eliminated there.

Reply:
hmm... maybe we *don't* have any pass that removes useless assignments? That should be a fairly easy pass... just walk every block backwards and remember any `StorageDead`. As long as you don't see a read from the locals that have a `StorageDead`, then you can remove assignments to the local.

Reply:
I actually did see a result of the pass (SimplifyLocals I think) that does what you say - if there is a variable that is just Live - copy-assign- Dead then it gets removed. The problem is that such patters it not present in MIR by default and only emerges e.g. as a result of this pass

Reply:
Oh, I think there is a pass that does what you say (not a pass from this PR), but it's under opt-level = 4 and also under "unsound mir optimizations"